opm/grid/cpgrid/Entity.hpp

```diff
@@ -566,6 +574,38 @@ Dune::cpgrid::Entity<0> Dune::cpgrid::Entity<codim>::getOrigin() const
     }
 }
 
+template<int codim>
+Dune::cpgrid::Entity<0> Dune::cpgrid::Entity<codim>::getLgrElem() const
+{
+    // Check that the element belongs to the leaf grid view
```

Comment:
If this only work for leaf elements then we should throw if the element is not on the leaf

Reply:
Exception is thrown in lines 586-588:
 else {
        throw std::invalid_argument("The entity provided does not belong to the leaf grid view. ");
    }

Reply:
Indeed there is an exception. 

To me it was not clear that `!(pgrid_ -> leaf_to_level_cells_.empty())` is the check that we are on the leaf grid view. Isn't this checking more than that? It seems like for a grid with no refinement (only one level) `pgrid_ -> leaf_to_level_cells_` would be empty too.

If that is intentional then we should probably adjust the comment and the message of the exception